templates/types/streaming/fastapi/app/api/routers/models.py

```diff
@@ -167,45 +167,39 @@ class SourceNodes(BaseModel):
     score: Optional[float]
     text: str
     url: Optional[str]
-    use_remote_file: bool = Field(
-        default=False,
-    )
 
     @classmethod
     def from_source_node(cls, source_node: NodeWithScore):
         metadata = source_node.node.metadata
-        url = metadata.get("URL")
+
         pipeline_id = metadata.get("pipeline_id")
         file_name = metadata.get("file_name")
         is_private = metadata.get("private", "false") == "true"
-        use_remote_file = metadata.get("use_remote_file", False)
-
-        if pipeline_id and file_name:
-            # If the nodes use data from LlamaCloud, we'll use a local file path as file name
-            file_name = f"{pipeline_id}${file_name}"
-            use_remote_file = True
-
-        # Construct file url through file server
-        if url is not None or file_name is not None:
-            url_prefix = os.getenv("FILESERVER_URL_PREFIX")
-            if not url_prefix:
-                logger.warning(
-                    "Warning: FILESERVER_URL_PREFIX not set in environment variables"
-                )
-            if file_name and url_prefix:
-                if use_remote_file:
-                    url = f"{url_prefix}/output/llamacloud/{file_name}"
-                elif is_private:
-                    url = f"{url_prefix}/output/uploaded/{file_name}"
+        is_local_file = metadata.get("is_local_file")
+        url_prefix = os.getenv("FILESERVER_URL_PREFIX")
+        if not url_prefix:
+            logger.warning(
+                "Warning: FILESERVER_URL_PREFIX not set in environment variables"
+            )
+        if file_name and url_prefix:
+            if not is_local_file:
+                if pipeline_id is None:
+                    logger.warning(
+                        "Warning: The file source is llamacloud but pipeline_id is not set. Cannot construct file url"
+                    )
```

Comment:
1. when does this case happen?
2. if it would not happen, we can remove `is_local_file`, correct?

Reply:
1. I don't think this would have happened if the data was consistent. But if it have we should not construct url and just return a null url to the FE and show a warning log in BE instead.
2. `is_local_file` mean that the file was stored locally or remotely (llama cloud), we need a flag to identify where the file is. Please let me know if you have better approach for this